crates/proto/src/rr/rdata/soa.rs

```diff
@@ -215,6 +217,22 @@ impl SOA {
     }
 }
 
+impl RecordData for SOA {
```

Comment:
I just started looking at this and I'm already getting confused about `RData` vs `RecordData` naming. Not sure about a better alternative... But at least should also add some documentation about when to use which.

Reply:
I think I still need to do this.

Reply:
I think one thing here is that RData could be unnecessary now, and we could move to `Box<dyn RecordData>` though I think I want to hold off on that in this change, it's something we could revisit in the future?